plugin/error_vis.py

```diff
@@ -29,17 +29,20 @@ def generate(self, view, errors):
             view (sublime.View): current view
             errors (list): list of parsed errors (dict objects)
         """
-        log.debug(" generating error regions for view %s", view.buffer_id())
-        # first clear old regions
-        if view.buffer_id() in self.err_regions:
-            log.debug(" removing old error regions")
-            del self.err_regions[view.buffer_id()]
-        # create an empty region dict for view id
-        self.err_regions[view.buffer_id()] = {}
-
-        for error in errors:
-            self.add_error(view, error)
-        log.debug(" %s error regions ready", len(self.err_regions))
+        try:
```

Comment:
Are you trying to catch TypeError exception that add_error() might throw or something else. You should be more specific in both which exception are you trying to catch and which code you expect to throw.

If you expect some method of view to throw then just wrap that call in a try/catch block.